src/Test/Utilities/Portable/Platform/Desktop/RuntimeAssemblyManager.cs

```diff
@@ -183,7 +183,12 @@ public bool HasConflicts(List<RuntimeModuleDataId> moduleDataIds)
         {
             foreach (var id in moduleDataIds.Select(x => x.Id))
             {
-                if (TryGetMatchingByFullName(id, out var assemblyData, out var fullMatch) && !fullMatch)
+                if (TryGetMatchingByFullName(id, out _, out var fullMatch) && !fullMatch)
+                {
+                    return true;
+                }
+
+                if (TryGetMatchingByMvid(id, out _, out fullMatch) && !fullMatch)
```

Comment:
Why should we accept an assembly that has the same MVID but not the same module ID?

Reply:
We don't. It's a match but not a full one.